### PR TITLE
update(blog)!: State of the Web App

### PR DESCRIPTION
**Pull request by friadev**
List of changes proposed in this PR:

- Added an article I wrote about IWAs

### CODE REVIEW

File: blog/posts/pwa-vs-iwa.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+## Improving Web Apps
+
+A typical native app is downloaded onto your computer from some kind of trusted place like an app store and only updates when the developers push an update out, and there's usually a process of checks and verification before that happens, like Apple's [App Review](https://developer.apple.com/distribute/app-review/) and the Google Play [review process](https://support.google.com/googleplay/android-developer/answer/9859455?hl=en). Plus it's much more difficult to only target a specific person, like someone with access to the servers that serve the HTML, CSS, and Javascript could do. Trying to push a malicious update to all users is more likely to get caught than a highly targeted attack.
```
Comment (redoomed1) [inline blog/posts/pwa-vs-iwa.md:26]:
```suggestion
A typical native app is downloaded onto your computer from some kind of trusted place like an app store and only receives updates when the developers push them out. Additionally, there's usually a process of checks and verification before that happens, like Apple's [App Review](https://developer.apple.com/distribute/app-review) and the Google Play [App Review](https://support.google.com/googleplay/android-developer/answer/9859455) process. In contrast to PWAs, with which the threat of an attacker with server access constantly looms, it's much more difficult to target a particular person. In other words, a malicious app update is much less likely to escape scrutiny than a highly targeted attack via compromised servers.
```

File: blog/posts/pwa-vs-iwa.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Isolated Web Apps (IWA) build on the work done on PWAs and [Web Packaging](https://github.com/WICG/webpackage). They are a specification that allows web content to be distributed offline outside of a browser, much like a traditional app. It can be signed just like a regular app too, allowing you to verify that it came from the proper place and hasn't been modified. You could install an IWA from your favorite app store just like any other app and have the same security assurances. This would be incredibly useful in allowing for cross-platform E2EE web apps that don't need to trust a server every time you use them.
+
+[Google](https://chromeos.dev/en/web/isolated-web-apps) distinguishes between the drive by web, PWAs, and IWAs. The drive by web requires more conservative access to the system as the most accessible and therefore least trusted option. PWAs are a bit more trusted and so can integrate a bit more deeply into the system. IWAs are the most trusted and as such can have deeper access into the system and more powerful capabilities.
```
Comment (redoomed1) [inline blog/posts/pwa-vs-iwa.md:30]:
```suggestion
Google [distinguishes](https://chromeos.dev/en/web/isolated-web-apps) between the drive by web, PWAs, and IWAs. The drive by web requires more conservative access to the system as the most accessible and is therefore least trusted. PWAs are a bit more trusted and can integrate a bit more deeply into the system as a result. IWAs are the most trusted and, as such, can have deeper access into the system and more powerful capabilities.
```

File: blog/posts/pwa-vs-iwa.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+<figure markdown>
+  ![Diagram showing the drive by web, a PWA, then a dotted line separating them from IWAs](../assets/images/iwa-the-future-of-web-apps/iwa-diagram.png)
+  <figcaption>source: <a href="https://chromeos.dev/en/web/isolated-web-apps">Google</a></figcaption>
+</figure>
+
+This higher security assurance from isolated and signed web applications and the inherently more trusted nature of a natively installed app will allow for IWAs to safely have access to APIs that it wouldn't be safe to allow normal websites to access, like [Direct Sockets](https://github.com/WICG/direct-sockets/blob/main/docs/explainer.md).
```
Comment (redoomed1) [inline blog/posts/pwa-vs-iwa.md:37]:
```suggestion
This higher security assurance from isolated and signed web applications and the inherently more trusted nature of a natively installed app will allow for IWAs to safely access APIs which wouldn't be safe to allow normal websites to access, like [Direct Sockets](https://github.com/WICG/direct-sockets/blob/main/docs/explainer.md).
```

File: blog/posts/pwa-vs-iwa.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+IWAs use a totally new [URL scheme](https://github.com/WICG/isolated-web-apps/blob/main/Scheme.md) since they're not relying on HTTPS certificate authorities or DNS. They're totally isolated from each other and the web using enforced Content Security Policy and Cross-Origin Isolation, hence the name.
+
+## Issues
+
+The [Worldwide Web Consortium](https://www.w3.org) currently has an open issue on their GitHub for IWAs with some interesting discussions that are worth checking out. There are some [criticisms](https://github.com/w3ctag/design-reviews/issues/842#issuecomment-1989631915) of IWAs, at least in their current form. A big point of contention is giving IWAs access to more powerful features like raw TCP and UDP socket access, similar to what a natively installed app might be able to do, which Martin Thomson at Mozilla argues is dangerous even with user consent. Martin has a nice in-depth [article](https://lowentropy.net/posts/bundles/) on bundling web content that's worth checking out on their website. It'll be a long process of iterating on the design before a version of this idea that's secure and available across browsers.
```
Comment (redoomed1) [inline blog/posts/pwa-vs-iwa.md:43]:
```suggestion
The [Worldwide Web Consortium](https://www.w3.org) currently has an open issue on their GitHub for IWAs with some interesting discussions that are worth checking out. There are some [criticisms](https://github.com/w3ctag/design-reviews/issues/842#issuecomment-1989631915) of IWAs, at least in their current form. A big point of contention is giving IWAs access to more powerful features like raw TCP and UDP socket access, similar to what a natively installed app might be able to do, which Martin Thomson at Mozilla argues is dangerous even with user consent. Martin wrote a nice in-depth [article](https://lowentropy.net/posts/bundles) on bundling web content that's worth checking out on their website. It'll be a long process of iterating on the design before a version of this idea that's secure and available across browsers.
```

File: blog/posts/pwa-vs-iwa.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+Right now, Chrome ships the feature [enabled by default](https://chromestatus.com/feature/5146307550248960) but only on ChromeOS for admin-controlled machines and select development partners of Google. Safari and Firefox haven't implemented the feature, with [Firefox](https://github.com/mozilla/standards-positions/issues/799#issuecomment-2342084330) taking a stance against it. Perhaps in its trial run the technology will prove its potential, or maybe IWAs aren't the best solution after all and another attempt will come along. I'll be watching with great interest either way.
```
Comment (redoomed1) [inline blog/posts/pwa-vs-iwa.md:45]:
```suggestion
Right now, Chrome ships the feature [enabled by default](https://chromestatus.com/feature/5146307550248960) but only on ChromeOS for admin-controlled machines and select development partners of Google. Safari and Firefox haven't implemented the feature, with [Firefox](https://github.com/mozilla/standards-positions/issues/799#issuecomment-2342084330) taking a stance against it. Perhaps in its trial run, the technology will prove its potential, or maybe IWAs aren't the best solution after all and another attempt at improving web apps will come along. I'll be watching with great interest either way.
```

File: blog/posts/pwa-vs-iwa.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    - Opinion
+authors:
+    - fria
+tags:
+    - PWA
+    - IWA
+    - Web
+license: BY-SA
+---
+# IWA: The Future of Web Apps?
```
[the remark this answers is not in the document]
Comment (redoomed1) [reply blog/posts/pwa-vs-iwa.md:14]:
```suggestion
# State of the Web App: Current Woes and Promising Futures
```
Offering this as an alternative title that attempts to capture the content of the article

[the remark this answers is not in the document]
Comment (friadev) [reply blog/posts/pwa-vs-iwa.md:14]: I like it

**Review by blacklight447 — APPROVED** [review]

**Review by redoomed1 — COMMENTED** [review]
These are more minor style suggestions.

File: blog/posts/pwa-vs-iwa.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Isolated Web Apps (IWA) build on the work done on PWAs and [Web Packaging](https://github.com/WICG/webpackage). They are a specification that allows web content to be distributed offline outside of a browser, much like a traditional app. It can be signed just like a regular app too, allowing you to verify that it came from the proper place and hasn't been modified. You could install an IWA from your favorite app store just like any other app and have the same security assurances. This would be incredibly useful in allowing for cross-platform E2EE web apps that don't need to trust a server every time you use them.
+
+Google [distinguishes](https://chromeos.dev/en/web/isolated-web-apps) between the drive by web, PWAs, and IWAs. The drive by web requires more conservative access to the system as the most accessible and is therefore least trusted. PWAs are a bit more trusted and can integrate a bit more deeply into the system as a result. IWAs are the most trusted and, as such, can have deeper access into the system and more powerful capabilities.
```
[the remark this answers is not in the document]
Comment (friadev) [reply blog/posts/pwa-vs-iwa.md:30]: I was copying the way google say drive by web in their article

File: blog/posts/pwa-vs-iwa.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+<figure markdown>
+  ![Diagram showing the drive by web, a PWA, then a dotted line separating them from IWAs](../assets/images/iwa-the-future-of-web-apps/iwa-diagram.png)
+  <figcaption>source: <a href="https://chromeos.dev/en/web/isolated-web-apps">Google</a></figcaption>
+</figure>
+
+This higher security assurance from isolated and signed web applications and the inherently more trusted nature of a natively installed app will allow for IWAs to safely access APIs which wouldn't be safe to allow normal websites to access, like [Direct Sockets](https://github.com/WICG/direct-sockets/blob/main/docs/explainer.md).
+
+IWAs use a totally new [URL scheme](https://github.com/WICG/isolated-web-apps/blob/main/Scheme.md) since they're not relying on HTTPS certificate authorities or DNS. They're totally isolated from each other and the web using enforced Content Security Policy and Cross-Origin Isolation, hence the name.
```
[the remark this answers is not in the document]
Comment (friadev) [reply blog/posts/pwa-vs-iwa.md:39]: They capitalize Cross-Origin Isolation in the documentation so I was copying that.

File: blog/posts/pwa-vs-iwa.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    - fria
+tags:
+    - PWA
+    - IWA
+    - Web
+license: BY-SA
+---
+# State of the Web App: Current Woes and Promising Futures
+
+The concept of a [progressive web app](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps) is enticing: an application using web technologies that is inherently cross platform (since it runs in a browser) and acts like a native app, even functioning offline. Support for PWAs in traditionally locked-down platforms like iOS means that PWAs can give users the freedom to install apps without having to go through Apple’s App Store. But there are problems with web content that PWAs haven't solved.<!-- more -->
```
[the remark this answers is not in the document]
Comment (friadev) [reply blog/posts/pwa-vs-iwa.md:16]: Not sure I like this wording as much, I think I'd rather leave it be.

File: blog/posts/pwa-vs-iwa.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+Google [distinguishes](https://chromeos.dev/en/web/isolated-web-apps) between the drive by web, PWAs, and IWAs. The drive by web requires more conservative access to the system as the most accessible and is therefore least trusted. PWAs are a bit more trusted and can integrate a bit more deeply into the system as a result. IWAs are the most trusted and, as such, can have deeper access into the system and more powerful capabilities.
+
+<figure markdown>
+  ![Diagram showing the drive by web, a PWA, then a dotted line separating them from IWAs](../assets/images/iwa-the-future-of-web-apps/iwa-diagram.png)
+  <figcaption>source: <a href="https://chromeos.dev/en/web/isolated-web-apps">Google</a></figcaption>
```
[the remark this answers is not in the document]
Comment (friadev) [reply blog/posts/pwa-vs-iwa.md:34]: same here I want to keep the same style from the Google article.

[the remark this answers is not in the document]
Comment (redoomed1) [reply blog/posts/pwa-vs-iwa.md:34]: I removed that suggestion, but I think that the caption should still be changed to specify the domain of the source (see the example from the [MPR blog post](https://www.privacyguides.org/articles/2024/11/17/where-are-all-the-mprs/#a-solution-multi-party-relays)).

**Review by redoomed1 — APPROVED** [review]
LGTM

**Pull request merged**